parlai/agents/bert_ranker/crossencoder_ranker.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+from parlai.core.torch_ranker_agent import TorchRankerAgent
+from .bert_dictionary import BertDictionaryAgent
+from .helpers import get_bert_optimizer, BertWrapper, BertModel, \
+    add_common_args, surround
+import torch
+import json
+
+
+class CrossEncoderRankerAgent(TorchRankerAgent):
+    """ TorchRankerAgent implementation of the crossencoder.
+        It is a standalone Agent. It might be called by the Both Encoder.
+    """
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        add_common_args(parser)
+
+    def __init__(self, opt, shared=None):
+        opt['rank_candidates'] = True
+        opt["bert_id"] = 'bert-base-uncased'
+        super().__init__(opt, shared)
+        # NOTE: This is done AFTER init so that it's after load(), on purpose.
+        # the state dict of MyModule and DataParallel(MyModule) is not the same
+        if self.opt["multigpu"]:
+            self.model = torch.nn.DataParallel(self.model)
```

Comment:
Hopefully dropped by #1319.

Reply:
for the written record, it's more complicated than that and we're keeping this for now.